nifi-nar-bundles/nifi-datadog-bundle/nifi-datadog-reporting-task/src/main/java/org/apache/nifi/reporting/datadog/DataDogReportingTask.java

```diff
@@ -0,0 +1,223 @@
+            .allowableValues(DATADOG_AGENT, DATADOG_HTTP)
+            .defaultValue(DATADOG_AGENT.getValue())
+            .build();
+
+    static final PropertyDescriptor API_KEY = new PropertyDescriptor.Builder()
+            .name("API key")
+            .description("DataDog API key. If specified value is 'agent', local DataDog agent will be used.")
+            .expressionLanguageSupported(false)
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    static final PropertyDescriptor METRICS_PREFIX = new PropertyDescriptor.Builder()
+            .name("Metrics prefix")
+            .description("Prefix to be added before every metric")
+            .required(true)
+            .expressionLanguageSupported(false)
+            .defaultValue("nifi")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    static final PropertyDescriptor ENVIRONMENT = new PropertyDescriptor.Builder()
+            .name("Environment")
+            .description("Environment, dataflow is running in. " +
+                    "This property will be included as metrics tag.")
+            .required(true)
+            .expressionLanguageSupported(false)
+            .defaultValue("dev")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    private MetricsService metricsService;
+    private DDMetricRegistryBuilder ddMetricRegistryBuilder;
+    private MetricRegistry metricRegistry;
+    private String metricsPrefix;
+    private String environment;
+    private String statusId;
+    private String apiKey;
+    private ConcurrentHashMap<String, AtomicDouble> metricsMap;
+    private volatile VirtualMachineMetrics virtualMachineMetrics;
+
+    @OnScheduled
+    public void setup(final ConfigurationContext context) {
+        metricsService = getMetricsService();
+        ddMetricRegistryBuilder = getMetricRegistryBuilder();
+        metricRegistry = getMetricRegistry();
+        metricsMap = getMetricsMap();
+        metricsPrefix = METRICS_PREFIX.getDefaultValue();
+        environment = ENVIRONMENT.getDefaultValue();
+        apiKey = API_KEY.getDefaultValue();
+        virtualMachineMetrics = VirtualMachineMetrics.getInstance();
+        ddMetricRegistryBuilder.setMetricRegistry(metricRegistry)
+                .setName("nifi_metrics")
+                .setTags(Arrays.asList("env", "dataflow_id"));
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        final List<PropertyDescriptor> properties = new ArrayList<>();
+        properties.add(METRICS_PREFIX);
+        properties.add(ENVIRONMENT);
+        properties.add(API_KEY);
+        properties.add(DATADOG_TRANSPORT);
+        return properties;
+    }
+
+    @Override
+    public void onTrigger(ReportingContext context) {
+        final ProcessGroupStatus status = context.getEventAccess().getControllerStatus();
```

Comment:
@JPercivall port and connection statuses in this group are empty, so could you please advice how can these statuses be retrieved from this place? Thanks.